### PR TITLE
Add domain transfer variant flow

### PR DESCRIPTION
**Pull request by agrullon95**
Fixes 3279-gh-Automattic/dotcom-forge

## Proposed Changes

This adds a variant of domain transfer flow, called `google-transfer`. It's tailored around Google Domains users.

### Original flow
1. The original flow should remain working as expected and unchanged.
2. Visit /setup/domain-transfer and go through the first two steps (not counting auth). 
3. If these work well, that's enough, the rest is untouched.
4. In the intro step, the pill in the "Checkout" **step should say "Free for Google Domains"**, and the explanation should be "Review your payment and contact details. If you're transferring a domain from Google, we'll pay for an additional year of registration.".

### New flow
1. Visit /setup/google-transfer.
2. The intro step header should be "Transfer your Google domains".
3. The subtitle should be "Follow these three simple steps to transfer your Google domains to WordPress.com.".
6. Under "Unlock your domains", there should a link leading to the instructions modal.
7. The instructions modal should look be eligible and look good on Mobile and Desktop.
4. In the intro step, the "Checkout" **step should say "We pay the first year".** 
5. In the intro step, the pill in the "Checkout" **step should say "We pay the first year"**, and the explanation should be "Review your payment and contact details. Google Domains transfers and the first year are free.".
### Preview

<table>
<tr>
 <td>

**Domain transfer**

</td>
 <td>

**Google Transfer**

 </td>
<tr>
 <td>

image

 </td>
 <td>

image

 </td>
</table>

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domains.tsx [hunk 1]
```diff
@@ -200,7 +200,7 @@ const Domains: React.FC< Props > = ( { onSubmit } ) => {
 			);
 		}
 
-		return __( 'Start transfer' );
```
Comment (ebinnion) [inline client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domains.tsx]: This was intentionally changed yesterday in a PR from @allilevine, so we should correct this.

File: client/landing/stepper/declarative-flow/internals/components/google-domains-transfer-instructions/style.scss [hunk 2]
```diff
@@ -0,0 +1,26 @@
+
```
Comment (ebinnion) [inline client/landing/stepper/declarative-flow/internals/components/google-domains-transfer-instructions/style.scss]: Some extra newlines here.

File: client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-intro/index.tsx [hunk 3]
```diff
@@ -9,16 +9,18 @@ import type { Step } from '../../types';
...
 	const { __ } = useI18n();
-	const isEnglishLocale = useIsEnglishLocale();
 
 	usePresalesChat( 'wpcom' );
 
 	const handleSubmit = () => {
 		submit?.();
 	};
+
+	const isGoogleDomainsTransferFlow = GOOGLE_TRANSFER === variantSlug;
```
Comment (ebinnion) [inline client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-intro/index.tsx:22]: I like this variant logic much more 👌

**Review by ebinnion — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8633866

Thank you @agrullon95 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8633866).

**Pull request merged**